src/multitask_personalization/envs/pybullet/pybullet_missions.py

```diff
@@ -0,0 +1,151 @@
+        self._llm = llm
+        self._seed = seed
+
+    def get_id(self) -> str:
+        return "book handover"
+
+    def get_mission_command(self) -> str:
+        # Could add some variation with an LLM later.
+        return "Please bring me a book to read"
+
+    def check_initiable(self, state: PyBulletState) -> bool:
+        return state.held_object is None
+
+    def check_complete(self, state: PyBulletState, action: PyBulletAction) -> bool:
+        robot_indicated_done = np.isclose(action[0], 2)
+        return bool(robot_indicated_done)
+
+    def step(
+        self, state: PyBulletState, action: PyBulletAction
+    ) -> tuple[str | None, float]:
+        robot_indicated_done = np.isclose(action[0], 2)
+        # Robot needs to indicate done for the handover task.
+        if not robot_indicated_done:
+            return None, 0.0
+        # Must be holding a book.
+        if state.held_object not in self._book_descriptions:
+            return None, -1.0
+        book_description = state.held_object
+        # Check if the book is reachable.
+        end_effector_position = self._robot.get_end_effector_pose().position
+        reachable = self._rom_model.check_position_reachable(
+            np.array(end_effector_position)
+        )
+        if not reachable:
+            return "I can't reach there", -1.0
+        # Should be holding a preferred book.
+        if not user_would_enjoy_book(
+            book_description,
+            self._hidden_book_preferences,
+            self._llm,
+            seed=self._seed,
+        ):
+            # The robot is attempting to hand over a book, but the user
+            # doesn't actually like that book. Have the user explain in
+            # natural language why they don't like the book.
+            text = _explain_user_book_preference(
+                book_description,
+                self._hidden_book_preferences,
+                self._llm,
+                enjoyed=False,
+                seed=self._seed,
+            )
+            return text, -1.0
+        # The robot is successful in handing over the book. Have the user
+        # elaborate on why they like this book.
+        text = _explain_user_book_preference(
+            book_description,
+            self._hidden_book_preferences,
+            self._llm,
+            enjoyed=True,
+            seed=self._seed,
+        )
+        return text, 1.0
+
+
+class StoreHeldObjectMission(PyBulletMission):
+    """Put away the thing the robot is holding."""
+
+    def get_id(self) -> str:
+        return "store held object"
+
+    def get_mission_command(self) -> str:
+        # Could add some variation with an LLM later.
+        return "Put away the thing you're holding"
+
+    def check_initiable(self, state: PyBulletState) -> bool:
+        return state.held_object is not None
+
+    def check_complete(self, state: PyBulletState, action: PyBulletAction) -> bool:
```

Comment:
I'm assuming the full version of this mission should also check that the held object is stored at the desired or appropriate place for `check_complete`